### ISSUE TITLE
[bug] Incorrect computation of td.large-*

### ISSUE THREAD

**Issue by allenwoot** [foundation/foundation-emails#186]
`td.large-*` math seems to be wrong: It also doesn't take into account the `$gutter_size` in computations
- For `center`, I think it should be `($i * ($width - 2*$gutter_size)/$num_columns)`
- For `non-center`, I'm not entirely sure what the intended effect is, but similarly it needs to account for gutters on the containing element

**Comment by kball** [comment]
Yeah, the math in here is odd, I agree, and definitely appears to be off.  The odd thing is it looks like we used to be accounting for gutters, but it changed it in this commit:  https://github.com/zurb/foundation-emails/commit/28cd0f86c16c29b468cb1e169548711468dc253d

@tdhartwick do you remember what that commit was fixing?  Am I likely to break something if I re-insert gutters into the computations?

**Comment by kball** [comment]
I tried implementing the math as described above, and it doesn't quite work out either, so lets see how this logic makes sense... Our current function looks like this:

```
// Calculates a pixel value for a grid column width.
/// @access private
/// @param {number} $columnNumber - Column count of the column.
/// @param {number} $totalColumns - Column count of the entire row.
/// @param {number} $containerWidth - Width of the surrounding container, in pixels.
/// @returns {number} A pixel width value.
@function -zf-grid-calc-px($columnNumber, $totalColumns, $containerWidth) {
  @return ($containerWidth / $totalColumns * $columnNumber - $global-gutter);
}
```

Or using the math in the issue report

```
@function -zf-grid-calc-px($columnNumber, $totalColumns, $containerWidth) {
  @return ($columnNumber * ($containerWidth - (2 * $global-gutter))/ $totalColumns);
}
```

Both the current version and the one in the issue report appear to make the numbers too big, with browsers then compensating in different ways.  So lets think about what this needs to be...

Hmm.  I think we've got a problem.  Basically, we've always got 2 gutters on the outside, but we need to account for anywhere from 0 to ($totalColumns - 1) additional gutters based on how many column elements there are... if we could do border-box sizing we wouldn't have to account for that, but that won't work in emails.  The workaround I can think of (which we could automate with inky) is to add a class that has the count of how many column elements there are, but otherwise I'm not seeing a way to compute this exactly right.

Am I missing something?  @tdhartwick @rafibomb @gakimball

**Comment by joeworkman** [comment]
Foundation for Emails has been superseded by [Inky v2](https://github.com/foundation/inky), a complete rewrite in Rust with improved Outlook support, hybrid output mode, bulletproof buttons, validation, and more. This issue will not be addressed in the legacy codebase.

If this issue persists in v2, please open a new issue at https://github.com/foundation/inky/issues
